File: core/core-domain/src/main/java/com/coffee/gu/cart/CartItemRepository.java
package com.coffee.gu.cart;


import com.coffee.gu.Principal;

import java.util.List;
import java.util.Optional;

public interface CartItemRepository {
    List<CartItem> findByPrincipalKey(String principalKey);
    Optional<CartItem> findByPrincipalIncludingDeleted(String principalKey);

    CartItem save(CartItem cartItem, Principal principal);
}

File: core/core-domain/src/main/kotlin/com/coffee/gu/cart/CartItemRepository.kt
package com.coffee.gu.cart

import com.coffee.gu.Principal

interface CartItemRepository {
    fun findByPrincipalKey(principalKey: String): List<CartItem>
    fun findByPrincipalKeyAndMenuIdIncludingDeleted(principalKey: String, menuId: Long): CartItem?
    fun save(cartItem: CartItem, principal: Principal): CartItem
}
